### DISCUSSION TITLE
Прошу обсудить построение справочника контрагентов

### DISCUSSION THREAD

**Discussion by LostAlly** [totumonline/totum-ru-issues-and-discussions#407] [Как это сделать?] [answered]
Добрый день.
Интересно ваше мнение при создании справочника контрагентов по такой схеме:
1. Таблица **контрагенты** используется для всех нефинансовых документов, так как очень часто фирмы имеют по несколько физ лиц(как зарегистрированные как ИП или привязанные сотрудники) и юрлиц
2. Справочник юр. лиц с реквизитами на самом деле и ип и физлица)
3. Справочник физлиц тут хранятся по сути дела контакты
2 и 3 связаны с 1 один ко многим
Далее везде где нет финансовых отношений с указанием реквизитов мы указываем контрагента из 1 и контактное лицо из 3
В документах где фигурируют финансы дополнительно указывается субъект из справочника юрлиц
Тут возникает ситуация когда люди могут находится в двух таблицах и в 1 и во 2 например как самозанятый или ип и как физ лицо.
Мне кажется это не очень хорошо.
Покритикуйте пожалуйста подход, боюсь нарваться на какой нибудь подводный камень с такой схемой или может можно сделать проще, но при это не терять связность.

**Comment by erpic-ru** [comment, marked as answer]
Делаю так: все что имеет под собой какие-то фин. обязательства и взаиморасчеты (Сотрудники, ИП, ООО) выношу в один справочник с выбором типа контрагента. Грубо говоря, все, что используется в проводках.

Все, что вокруг контрагентов, например, несколько контактов, адресов, расчетных счетов или ещё чего-то идет связанными справочниками

**Reply by ttm-alexeypolunin** [reply]
Ну тут вы для конкретного случая выбираете баланс между сложностью и подробностью информации — нет никакого железного варианта.

Обычно Контрагент = ИНН (и ИНН делается уникальным и обязательным, что бы дубли не создавать). Если он Контрагент физик, то у него в этой же таблице признак Физик и тогда без ИНН и его обязательности.

Контакты часто просто делаются текстовым полем, а иногда строками внутри цикла если Контрагенты — это таблица циклов.

Если вам хочется сквозную таблицу с контактами — ок. Можете тогда для физиков Контрагентов сделать привязку к этим контактам, но мне кажется ее не будут делать.
